posthog/queries/util.py

```diff
@@ -23,44 +23,51 @@ def parse_timestamps(filter: FilterType, team: Team, table: str = "") -> Tuple[s
     date_to = None
     params = {}
     if filter.date_from:
-
-        date_from = f"AND {table}timestamp >= toDateTime(%(date_from)s, %(timezone)s)"
-        params.update({"date_from": format_ch_timestamp(filter.date_from, filter, timezone=team.timezone_for_charts)})
+        date_from = f"AND {table}timestamp >= toDateTime(%(date_from)s)"
+        params.update(
+            {
+                "date_from": format_ch_timestamp(
+                    filter.date_from,
+                    convert_to_timezone=team.timezone_for_charts if not filter.date_from_has_explicit_time else None,
+                )
+            }
+        )
     else:
         try:
             earliest_date = get_earliest_timestamp(team.pk)
         except IndexError:
             date_from = ""
         else:
-            date_from = f"AND {table}timestamp >= toDateTime(%(date_from)s, %(timezone)s)"
-            params.update({"date_from": format_ch_timestamp(earliest_date, filter)})
+            date_from = f"AND {table}timestamp >= toDateTime(%(date_from)s)"
+            params.update({"date_from": format_ch_timestamp(earliest_date)})
 
     _date_to = filter.date_to
 
-    date_to = f"AND {table}timestamp <= toDateTime(%(date_to)s, %(timezone)s)"
-    params.update({"date_to": format_ch_timestamp(_date_to, filter, " 23:59:59")})
+    date_to = f"AND {table}timestamp <= toDateTime(%(date_to)s)"
+    params.update(
+        {
+            "date_to": format_ch_timestamp(
+                _date_to,
+                convert_to_timezone=team.timezone_for_charts
+                if filter._date_to and not filter.date_to_has_explicit_time
+                else None,
+            )
+        }
+    )
 
     return date_from or "", date_to or "", params
 
 
-def format_ch_timestamp(
-    timestamp: datetime, filter, default_hour_min: str = " 00:00:00", timezone: Optional[str] = None
-):
-    if timezone:
+def format_ch_timestamp(timestamp: datetime, convert_to_timezone: Optional[str] = None):
```

Comment:
Was the timezone issue double conversion? As in, the timezone is handled here and then handled once again in `toDateTime`

Reply:
Yeah partly, when I did it before it was inconsistent which led to issues. Now we'll just always do it here.